binderhub/static/js/index.js

```diff
@@ -52,67 +52,67 @@ function build(providerSpec, log, fitAddon, path, pathType) {
     buildToken,
   );
 
-  image.onStateChange("*", function (data) {
+  for await (const data of image.fetch()) {
+    // Write message to the log terminal if there is a message
     if (data.message !== undefined) {
       log.writeAndStore(data.message);
       fitAddon.fit();
     } else {
       console.log(data);
     }
-  });
-
-  image.onStateChange("waiting", function () {
-    $("#phase-waiting").removeClass("hidden");
-  });
-
-  image.onStateChange("building", function () {
-    $("#phase-building").removeClass("hidden");
-    log.show();
-  });
-
-  image.onStateChange("pushing", function () {
-    $("#phase-pushing").removeClass("hidden");
-  });
-
-  image.onStateChange("failed", function () {
-    $("#build-progress .progress-bar").addClass("hidden");
-    $("#phase-failed").removeClass("hidden");
-
-    $("#loader").addClass("paused");
 
-    // If we fail for any reason, show an error message and logs
-    updateFavicon(BASE_URL + "favicon_fail.ico");
-    log.show();
-    if ($("div#loader-text").length > 0) {
-      $("#loader").addClass("error");
-      $("div#loader-text p.launching").html(
-        "Error loading " + spec + "!<br /> See logs below for details.",
-      );
+    switch (data.phase) {
+      case "waiting":
```

Comment:
Apparently the content under case should be put in a curly bracket

https://docs.w3cub.com/eslint/rules/no-case-declarations

Reply:
oooh, fun lexical scope thingies. I'll add.

Reply:
Fixed